blackjack/blackjack.py

```diff
@@ -0,0 +1,217 @@
+                player.set_hand_status("S")
+            elif player.get_hand_value() + card.get_value() > 21:
+                player.set_status("B")
+    else:
+        if player.get_hand_value() + card.get_value() > 21:
+            if player.get_hand_status() == "S":
+                if player.get_hand_value() + card.get_value() - 10 <= 21:
+                    player.set_hand_status("H")
+                else:
+                    player.set_status("B")
+            else:
+                player.set_status("B")
+
+    player.add_card(card)
+    return player
+
+# initialize adds the first 2 cards to player and dealer hands
+# if possible to add a A with a value of 11, it does so by setting
+# the hand to Soft. If not, the hand is Hard and the Ace will have
+# a value of 1
+
+
+def initialize(player, card1, card2):
+    # Initial 2 cards to dealer
+    player.add_card(card1)
+    player.add_card(card2)
+    player.set_hand_status("H")
+    if card1.get_rank() == "A" or card2.get_rank() == "A":
+        if card1.get_value() + card2.get_value() + 10 <= 21:
+                player.set_hand_status("S")
+    return player
+
+# dealer_plays manages a loop for the dealer dealing a new card,
+# adding it to the hand, as long as the value of the hand is
+# smaller than 17
+
+
+def dealer_plays(mydealer):
+    mydealer.set_status('H')
+    print("Finally my turn!!!")
+    print('Here I go')
+    while mydealer.get_hand_value() < 17:
+        print("My current hand is: {}".format(mydealer.get_hand()))
+        d_card = mydealer.deal()
+        mydealer.set_hand_status("H")
+        print("My new card is {}".format(d_card))
+        mydealer = validate_card(mydealer, d_card)
+
+# player_plays manages a loop for the player dealing a new card
+# adding it to the hand, and asking the use if they want to hit or stand
+
+
+def player_plays(myplayer, mydealer):
+    myplayer.set_status('H')
+    while myplayer.get_status() == "H":
+        print("Your current hand is: {}".format(myplayer.get_hand()))
+        while True:
+            action = input("Hit (H) or Stand? (S) ").upper()
+            if action in ["H", "S"]:
+                break
+        myplayer.set_status(action)
+        if myplayer.get_status() != "S":
+            p_card = mydealer.deal()
+            print("Your card is {}".format(p_card))
+            myplayer = validate_card(myplayer, p_card)
+            if myplayer.get_hand_value() == 21:
+                myplayer.set_status('S')
+
+# print status lets the player knows how many games they have won, lost or tied
+
+
+def print_status(game):
+    print('--------------')
+    print("GLOBAL SCORES")
+    print('You have won {} games'.format(game.get_winners().count("P")))
+    print('I have won {} games'.format(game.get_winners().count("D")))
+    print('And we have Tied {} times'.format(game.get_winners().count("T")))
+    print('--------------')
+
+# main program loop.
+
+
+def play_game():
```

Comment:
This function should be approximately 5-7 functions. If you want to go over how to refactor a giant function like this, let's pair on it sometime.